File: Src/Icm.ContextConsole/NamesSynonyms/INamedWithSynonymsExtensions.vb
Imports System.Runtime.CompilerServices

Public Module INamedWithSynonymsExtensions

    <Extension>
    Public Function IsNamed(obj As INamedWithSynonyms, name As String) As Boolean
        Dim lowname = name.ToLower
        Return obj.Name = lowname OrElse obj.Synonyms.Contains(lowname)
    End Function

End Module

File: Src/Icm.ContextConsole/NamesSynonyms/INamedWithSynonymsExtensions.cs
using System.Linq;

public static class INamedWithSynonymsExtensions
{
	public static bool IsNamed(this INamedWithSynonyms obj, string name)
	{
		var lowname = name.ToLower();
		return obj.Name() == lowname || obj.Synonyms().Contains(lowname);
	}

}
